protomongo/protomongo.go

```diff
@@ -167,7 +173,11 @@ func (pc *protobufCodec) protoHelper(pb descriptor.Message, t reflect.Type) *pro
 		oneofPropsByTag[strconv.Itoa(oneof.Prop.Tag)] = oneof
 	}
 	ph := &protoHelper{normalPropsByTag, oneofPropsByTag, oneofFieldWrapperProps}
+
+	pc.mux.Lock()
 	pc.protoHelpers[messageName] = ph
+	pc.mux.Unlock()
```

Comment:
this isn't quite correct - if code between `Lock()` and `Unlock()` returns early (or panics), the mutex will never unlock. we should really `defer pc.mux.Unlock()` straight after locking it.

Reply:
Ah very cool